include/pf-applications/base/data.h

```diff
@@ -0,0 +1,20 @@
+#include <array>
+#include <utility>
+
+namespace internal
+{
+  template <typename T, std::size_t... Is>
+  constexpr std::array<T, sizeof...(Is)>
+  create_array(T value, std::index_sequence<Is...>)
```

Comment:
Could you document what this function does? The code is utterly C++ code.

Reply:
For some reason Github loaded this comment after I merged this PR seeing the approval. I will then add comments in a separate PR.

This code simply create std::array of a given sizes with all of its elements initialized to `value`.